### PR TITLE
Refresh new partitions

### PR DESCRIPTION
**Pull request by williamorrie**

### CODE REVIEW

File: python_scripts/database_builder_v2.py [hunk 1]
```diff
@@ -605,21 +588,7 @@ def rebuild_database(db_name, rebuild_db, delete_underlying_data=True):
...
-    # glue_client.create_database(DatabaseInput={
-    #                                 'Name': db_name,
-    #                                 'Description': f'Database for storing tables extracted from the matrix booking system API: {args.env} environment',
-    #                                 'LocationUri': f'alpha-dag-matrix/db/{args.env}/'
-    #                                     }
-
-    #                            )
-
-    # Add tables
+    rebuild_database(database_name="matrix_db_dev", rename_db=db_name)
```
Comment (gwionap) [inline python_scripts/database_builder_v2.py:591]: Why is the database name hardcoded?

File: python_scripts/database_builder_v2.py [hunk 2]
```diff
@@ -630,4 +599,4 @@ def rebuild_database(db_name, rebuild_db, delete_underlying_data=True):
     # Joined rooms
     glue_client.create_table(**schema_joined_rooms)
 
-    rebuild_all_s3_data_from_raw()
+    #rebuild_all_s3_data_from_raw()
```
Comment (gwionap) [inline python_scripts/database_builder_v2.py:602]: Why is this commented out now?

File: python_scripts/database_builder_v2.py [hunk 3]
```diff
@@ -513,43 +503,48 @@ def delete_database_data(db_path):
...
+    Optionally rename database. 
+    
+    Args:
+        database_name (str): Name of current database
+        delete_data (bool): True if underlying data should be deleted
+        rename_db (str) New database name
     """
-
     # s3 client for handling exceptions
     s3 = boto3.client("s3")
```
Comment (gwionap) [inline python_scripts/database_builder_v2.py:522]: Does this function need some additional upfront logic i.e. if rename_db is None or rename_db == database_name then if database_name exists in glue catalog do nothing?

File: python_scripts/functions/data_validation.py [hunk 4]
```diff
@@ -220,13 +226,32 @@ def read_and_write_cleaned_data(
         logger.info(f"{name} data for {start_date} written to s3.")
 
 
+def refresh_new_partition(database_name: str, table_name: str, scrape_date: str):
+    athena = boto3.client("athena", region_name=region_name)
+    query_string = f"""alter table awsdatacatalog.{database_name}.{table_name} 
+                add partition (scrape_date = '{scrape_date}')"""
+    logger.info(f"Athena Query: adding {scrape_date} partition to \
+                {database_name}.{table_name}")
+    resp = athena.start_query_execution(QueryString=query_string)
```
Comment (gwionap) [inline python_scripts/functions/data_validation.py:235]: I think you need to wait for the query to finish to check it executed successfully. You can probably swap out the boto3 calls for `wr.athena.start_query_execution` and then run `wr.athena.wait_query` based on the returned execution id, from wrangler (see [here](https://aws-sdk-pandas.readthedocs.io/en/stable/api.html#amazon-athena)).

**Review by gwionap — CHANGES_REQUESTED** [review]
Thanks @williamorrie - couple of questions / suggestions below

File: python_scripts/database_builder_v2.py (hunk 1 above)
Comment (williamorrie) [reply python_scripts/database_builder_v2.py:591]: I could not figure out how to pass database_name as a separate variable. I was able to get it working enough to rename the dev db, I didn't mean to commit it in the state it was in.

File: python_scripts/database_builder_v2.py (hunk 2 above)
Comment (williamorrie) [reply python_scripts/database_builder_v2.py:602]: Same as above, I was only wanting to rename a database using the code that was already there, didn't think there was any benefit to recreating the data as it hadn't moved.

File: python_scripts/database_builder_v2.py (hunk 3 above)
Comment (williamorrie) [reply python_scripts/database_builder_v2.py:522]: Would it be easy to separate database_builder from this PR? I know that this function needs fixed, but don't want it to hold up automating the refresh of the schema.

File: python_scripts/functions/data_validation.py (hunk 4 above)
Comment (williamorrie) [reply python_scripts/functions/data_validation.py]: I knew it was lazy just returning the id, it took me an entire day to figure out that the lack of return value was the cause of the problem in Airflow and i thought it was better to submit some type of progress than waste more time.

I also couldn't understand why the `read_and_write_cleaned_data` function, which also had no return value, didn't cause a problem (TypeError: 'NoneType' object is not callable) when the function was assigned to a variable in main?

That nice that awswrangler raises an error on failed.

**Review by gwionap — APPROVED** [review]

**Pull request merged**